### PR TITLE
Resolving TODO

### PR DESCRIPTION
**Pull request by ricardoak25**
- Extract the StandByValidators from the configuration file
- In Endpoint.cs, perform a more safe parsing in Parse method
- Use singleton in GenesisAssets

### CODE REVIEW

File: src/NeoSharp.Core/Network/EndPoint.cs [hunk 1]
```diff
@@ -101,7 +101,13 @@ public override bool Equals(object obj)
 
         private static EndPoint Parse(string protocol, string address)
         {
-            if (protocol == null) protocol = DefaultProtocol;
+            var protocolEnum = Protocol.Unknown;
+            if (String.IsNullOrWhiteSpace(protocol)) {
+                protocolEnum = DefaultProtocol;
+            }
+            else if (protocol.Equals("tcp", StringComparison.InvariantCultureIgnoreCase)) {
```
Comment (shargon) [inline src/NeoSharp.Core/Network/EndPoint.cs]: Enum.TryParse?

**Comment by ricardoak25** [comment]
- In EndPoint.cs, check DNS when converting to IpEndPoint
- Check precision in WeightedFilter (Extensions.cs)
- In RocksDBContext, "avoid sync IO in constructor -> Open connection with the first operation for now"
- In ConfigurationExtensions.cs, ParseIpEndPoint should use a more safe parsing

File: src/NeoSharp.Core/Network/EndPoint.cs [hunk 2]
```diff
@@ -115,10 +122,12 @@ private static EndPoint Parse(string protocol, string address)
                 port = int.Parse(address.Substring(portIndex + 1));
             }
 
-            // TODO: More safe parsing
+            Protocol protocolEnum;
+            Enum.TryParse<Protocol>(protocol, out protocolEnum);
```
Comment (shargon) [inline src/NeoSharp.Core/Network/EndPoint.cs:126]: if not?

Comment (ricardoak25) [reply src/NeoSharp.Core/Network/EndPoint.cs:126]: Protocol.Unknow

Comment (shargon) [reply src/NeoSharp.Core/Network/EndPoint.cs:126]: Check the travis error, and i merge it

**Comment by ricardoak25** [comment]
- Unit test fixed.
- Rebase

File: src/NeoSharp.Core/Network/NetworkConfig.cs [hunk 3]
```diff
@@ -25,6 +26,10 @@ public class NetworkConfig
         /// Acl Config
         /// </summary>
         public NetworkAclConfig AclConfig { get; internal set; }
+        /// <summary>
+        /// StandByValidator config
+        /// </summary>
+        public IEnumerable<string> StandByValidator { get; internal set; }
```
Comment (shargon) [inline src/NeoSharp.Core/Network/NetworkConfig.cs]: string[] ?

Comment (ricardoak25) [reply src/NeoSharp.Core/Network/NetworkConfig.cs]: If you prefer, i can change to string[].
I generally prefer IEnumerable<T>

**Comment by ricardoak25** [comment]
- Value in Fixed8 should be readonly
- Create unit testing for SecureString conversions

**Review by shargon — APPROVED** [review]

**Pull request merged**